src/hooks/useApproveAndDeposit.ts

```diff
@@ -142,7 +143,9 @@ export function useApproveAndDeposit(
       const spendTransaction = await swapContract.addLiquidity(
         POOL_TOKENS.map(({ symbol }) => state.tokenFormState[symbol].valueSafe),
         minToMint,
-        Math.round(new Date().getTime() / 1000 + 60 * 10),
+        Math.round(
+          new Date().getTime() / 1000 + 60 * state.transactionDeadline,
```

Comment:
is `state.transactionDeadline` always guaranteed to be populated at this point?

Reply:
double checked and the logic for this is handled in the deadline field and works across all deadline fields regardless the page

found in DeadlineField.tsx
`onChange={(e): void => {
            const deadlineNumber =
              e.target.value === "" || parseFloat(e.target.value) <= 0
                ? Deadlines.Ten // if left blank, deadline defaults to 10
                : parseFloat(e.target.value)
            dispatch(updateTransactionDeadline(deadlineNumber))
          }}`